api/base/parsers.py

```diff
@@ -74,6 +75,14 @@ def flatten_data(self, resource_object, parser_context, is_list):
             if object_type is None:
                 raise JSONAPIException(source={'pointer': '/data/type'}, detail=NO_TYPE_ERROR)
 
+        if parser_context['request'].version >= 2.6:
```

Comment:
Has to be the next available version because it will break existing 2.6 code if we add it. We may see about releasing this with any other upcoming breaking changes when the time comes.